coderd/workspaceagents_test.go

```diff
@@ -1107,6 +1110,162 @@ func TestWorkspaceAgent_Metadata(t *testing.T) {
 	post("unknown", unknownKeyMetadata)
 }
 
+type testWAMErrorStore struct {
+	database.Store
+	err atomic.Pointer[error]
+}
+
+func (s *testWAMErrorStore) GetWorkspaceAgentMetadata(ctx context.Context, arg database.GetWorkspaceAgentMetadataParams) ([]database.WorkspaceAgentMetadatum, error) {
+	err := s.err.Load()
+	if err != nil {
+		return nil, *err
+	}
+	return s.Store.GetWorkspaceAgentMetadata(ctx, arg)
+}
+
+func TestWorkspaceAgent_Metadata_CatchMemoryLeak(t *testing.T) {
```

Comment:
Should this instead be a benchmark?

Reply:
I wanted to keep this as a test to verify the endpoint behaves as it should. A benchmark wouldn't help in catching a memory leak either since it requires being run manually and checking output (assuming it doesn't get OOM killed).